lua/dap/ui/variables.lua

```diff
@@ -32,14 +32,15 @@ local function write_variables(buf, variables, line, column, win)
   local win_height = api.nvim_win_get_height(win)
   local indent = string.rep(" ", column)
 
+  local function padzeros(d) return ("%010d"):format(d) end
   local sorted_variables = {}
   for _, v in pairs(variables) do
     table.insert(sorted_variables, v)
   end
   table.sort(
     sorted_variables,
     function(a, b)
-      return a.name < b.name
+      return a.name:gsub("^%[?(%d+)%]?$", padzeros) < b.name:gsub("^%[?(%d+)%]?$", padzeros)
```

Comment:
I had something like this in mind:


```suggestion
      local num_a = string.match(a.name, '^%[?(%d+)%]?$')
      local num_b = string.match(b.name, '^%[?(%d+)%]?$')
      if num_a and num_b then
        return tonumber(num_a) < tonumber(num_b)
      else
        return a.name < b.name
      end
```

It would avoid the arbitrary padding to 10.